src/js/hooks/list-pages/useTableData.jsx

```diff
@@ -43,8 +43,14 @@ const useTableData = ({
     // Each time we fetch, we want to 'reset' the token/signal
     sourceRef.current = CancelToken.source();
     // reset pagination on each search execution
-    tableRef.current.onPageChange(0);
-    tableRef.current.fireFetchData();
+    if (tableRef.current?.state?.page > 0) {
+      // onPageChange(pageIndex) triggers fireFetchData() when pageIndex !== currenPage
+      // which is why we are calling onPageChange(0) and fireFetchData() separately
+      // by doing that we are trying to avoid double fetching
+      tableRef.current.onPageChange(0);
+    } else {
+      tableRef.current.fireFetchData();
+    }
```

Comment:
a question (shout at me if I just don't understand the context), but if `onPageChange(0)` anyway triggers the `fireFetchData`, can't we just remove the `tableRef.current.fireFetchData()` from line 47th, and that would be it?

Reply:
It is being triggered by useEffect.
The useEffect is calling `fireFetchData()` on the initial load and when filters change.
https://github.com/openboxes/openboxes/blob/a4a07bf1ee4fe01e6699d41b884d8ed4dcbcd66c/src/js/hooks/list-pages/useTableData.jsx#L57